pennylane/configuration.py

```diff
@@ -55,7 +53,7 @@ def __init__(self, name):
                 break
             except FileNotFoundError:
                 if idx == len(directories) - 1:
-                    log.info("No PennyLane configuration file found.")
+                    warnings.warn("No PennyLane configuration file found.", UserWarning)
```

Comment:
Will this be visible on import of PL? I have a feeling that's why we had `logger.info` instead of `logger.warn` previously. @trbromley

Reply:
+1 not to have this show up on import. Can we limit the scope of the warning?

Reply:
Sorry for not mentioning this change. I saw your thumbs up in [this comment](https://github.com/PennyLaneAI/pennylane/issues/2006#issuecomment-1009413835) and assumed we could delete it.

This definitely raises an error when importing PL... I'll keep using `logging` then, but use our own logger instance.

Reply:
@trbromley maybe using a [ResourceWarning](https://docs.python.org/3/library/exceptions.html#ResourceWarning)?

This warning is only shown in Python Development Mode. My question is: will users ever use this mode? If they will (most probably) never see the warning, should we just remove it?

Reply:
Yeah, good point @AlbertMitjans. Another option is to use `log = logging.getLogger(__name__)` here, so that this module is not using or getting the root logger 🤔 Although I don't know if that has implications for the users reporting those bugs.

Reply:
I used [this user's suggestion](https://github.com/PennyLaneAI/pennylane/issues/3731#issue-1574552685): `log = logging.getLogger(__name__)`.

I'm not super familiar with `logging`, but will users ever see this `info` message? Don't we need a handler to display the log messages?

Reply:
FYI, I believe the `info` message was never shown to users, because the root logger is instantiated with level WARNING, thus all INFO messages are ignored:
```pycon
>>> import logging
>>> logging.info("hello world")  # nothing is shown/logged
```
Consequently, I believe we can definitely remove this info message without changing any code behaviour.